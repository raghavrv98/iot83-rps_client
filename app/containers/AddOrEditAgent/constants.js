/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
