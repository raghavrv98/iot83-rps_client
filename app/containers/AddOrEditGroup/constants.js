/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
