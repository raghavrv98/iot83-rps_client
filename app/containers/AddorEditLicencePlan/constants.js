/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";