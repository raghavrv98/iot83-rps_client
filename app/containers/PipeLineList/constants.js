/*
 *
 * PipeLineList constants
 *
 */

export const DEFAULT_ACTION = "app/PipeLineList/DEFAULT_ACTION";

export const GET_PIPELINE_LIST = "app/PipeLineList/GET_PIPELINE_LIST";
export const GET_PIPELINE_LIST_SUCCESS = "app/PipeLineList/GET_PIPELINE_LIST_SUCCESS";
export const GET_PIPELINE_LIST_FAILURE = "app/PipeLineList/GET_PIPELINE_LIST_FAILURE";

export const DELETE_PIPELINE = "app/PipeLineList/DELETE_PIPELINE";
export const DELETE_PIPELINE_SUCCESS = "app/PipeLineList/DELETE_PIPELINE_SUCCESS";
export const DELETE_PIPELINE_FAILURE = "app/PipeLineList/DELETE_PIPELINE_FAILURE";

export const GET_PLANT_LIST = "app/PipeLineList/GET_PLANT_LIST";
export const GET_PLANT_LIST_SUCCESS = "app/PipeLineList/GET_PLANT_LIST_SUCCESS";
export const GET_PLANT_LIST_FAILURE = "app/PipeLineList/GET_PLANT_LIST_FAILURE";

export const GET_ALARM_LIST = "app/PipeLineList/GET_ALARM_LIST";
export const GET_ALARM_LIST_SUCCESS = "app/PipeLineList/GET_ALARM_LIST_SUCCESS";
export const GET_ALARM_LIST_FAILURE = "app/PipeLineList/GET_ALARM_LIST_FAILURE";

export const GET_PIPELINE_SEGMENT_DATA = "app/PipeLineList/GET_PIPELINE_SEGMENT_DATA";
export const GET_PIPELINE_SEGMENT_DATA_SUCCESS = "app/PipeLineList/GET_PIPELINE_SEGMENT_DATA_SUCCESS";
export const GET_PIPELINE_SEGMENT_DATA_FAILURE = "app/PipeLineList/GET_PIPELINE_SEGMENT_DATA_FAILURE";