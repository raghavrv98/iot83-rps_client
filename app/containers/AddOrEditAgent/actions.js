/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
