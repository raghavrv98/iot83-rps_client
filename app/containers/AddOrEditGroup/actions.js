/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
