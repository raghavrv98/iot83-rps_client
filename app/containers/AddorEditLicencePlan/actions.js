/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}