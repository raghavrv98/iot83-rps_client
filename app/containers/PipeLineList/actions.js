/*
 *
 * PipeLineList actions
 *
 */

import {GET_PIPELINE_LIST,DELETE_PIPELINE,GET_PLANT_LIST, GET_ALARM_LIST, GET_PIPELINE_SEGMENT_DATA } from "./constants";

export function getPipelineList(id) {
  return {
    type: GET_PIPELINE_LIST,
    id
  };
}

export function pipelineDeleteHandler(plantId,pipeId) {
  return {
    type: DELETE_PIPELINE,
    plantId,pipeId
  };
}

export function managePlantList() {
  return {
    type: GET_PLANT_LIST,
  };
}

export function manageGetAlarmList(plantId, limit, offset) {
  return {
    type: GET_ALARM_LIST,
    plantId, 
    limit, 
    offset
  };
}
