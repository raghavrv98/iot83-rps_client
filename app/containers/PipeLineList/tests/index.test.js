import React, { Component } from "react";
import { mount, shallow } from "enzyme";
import history from "utils/history";
import { getPipelineList, pipelineDeleteHandler, managePlantList, manageGetAlarmList } from "../actions";
import { GET_PIPELINE_LIST, DELETE_PIPELINE, GET_PLANT_LIST, GET_ALARM_LIST } from "../constants";
import { PipeLineList, mapDispatchToProps } from "../index";
import { result } from "lodash";

const pipelineList = {
  _id: "9bed0d5ca79011e9863a0242ac110016",
  createdAt: 1563257514466,
  name: "edfedeefdf",
  other: "fvcerdfedfv",
  overallGeometry: "dfe3dfedc",
  plantId: "f29b79c09f0911e986810242ac110011",
  processFluid: "efe3dcfed",
  updatedAt: 1563257547048,
  result: []
}

const pipelineListFailure = "fail";
const pipeLineDelete = "delete";
const pipelineDeletedSuccess = "success";
const pipeLineDeleteFailure = "fail";
const plantList = {
  result: [{
    _id: "892d2858168d11ea8a030242ac120014",
    alarmCount: 0,
    createdAt: 1575460773585,
    name: "Unamboach_Plant",
    ownerEmail: "donnadrobinson@spy.com",
    ownerName: "Donna Robinson",
    ownerPhone: "+917775757579",
    pipelineCount: 0,
    plantImage: null,
    updatedAt: 1575461564520
  }],
  count: 1
}
const plantListFailure = "fail";
const getAlarmList = [];
const getAlarmListFailure = "fail"
describe("< PipeLineList />", () => {

  it('match snapshots', () => {
    const wrap = shallow(<PipeLineList match={match} pipelineDeleteHandler={pipelineDeleteHandler} getPipelineList={getPipelineList} managePlantList={managePlantList} manageGetAlarmList={manageGetAlarmList} />)
    expect(wrap).toMatchSnapshot()
  })

  const match = {
    params: {
      id: ""
    }
  };

  it('should test component when render', () => {
    spyOn(PipeLineList.prototype, 'render').and.callThrough();
    const wrapper = shallow(<PipeLineList getPipelineList={getPipelineList} pipelineDeleteHandler={pipelineDeleteHandler} list={PipeLineList} match={match} managePlantList={managePlantList} manageGetAlarmList={manageGetAlarmList} />);
    expect(wrapper).toBeDefined();
    expect(PipeLineList.prototype.render).toHaveBeenCalledTimes(1);
  })

  it("clicking back button trigger onClick event", async () => {
    const match = {
      params: {
        id: "demoId"
      }
    };
    let expectedCount = history.length + 1;
    const wrapper = shallow(
      <PipeLineList
        history={history}
        match={match}
        getPipelineList={getPipelineList}
        managePlantList={managePlantList}
        pipelineDeleteHandler={pipelineDeleteHandler}
        manageGetAlarmList={manageGetAlarmList}
      />
    );
    let backbutton = wrapper.find("button.btn-transparent").first();
    expect(backbutton.length).toBe(1);
    backbutton.simulate("click");
    expect(history.length + 1).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/");
  });

  // it("clicking detail button trigger onClick event", async () => {
  //   const match = {
  //     params: {
  //       id: "demoId"
  //     },
  //   };
  //   const fetchpipeLineList=[{
  //         _id: "8fdc4e0aac6811e994a00242ac110019"
  //     }]
  //   let expectedCount = history.length + 1;
  //   const wrapper = shallow(
  //     <PipeLineList
  //       history={history}
  //       match={match}
  //       pipelineList={fetchpipeLineList}
  //       getPipelineList={getPipelineList} 
  //     />
  //   );
  //   wrapper.setProps({ pipelineList: pipelineList});
  //   let viewDetails = wrapper.find("button.text-orange");
  //   viewDetails.simulate("click");
  //   expect(viewDetails.length).toBe(1);
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/pipeLineDetail/demoId/"+pipelineList[0]._id);
  // });
  // it("clicking show pipeline trigger onClick event", async () => {
  //     const match = {
  //         params: {
  //           id: "demoId"
  //         }
  //       };
  //     let expectedCount = history.length + 1
  //     const wrapper = shallow(
  //       <PipeLineList
  //         history={history}
  //         match={match}
  //         pipelineList=""
  //         getPipelineList={getPipelineList}
  //       />
  //     );
  //     wrapper.setProps({ pipelineList: pipelineList});
  //     let viewDetails = wrapper.find("button.text-success");
  //     expect(viewDetails.length).toBe(1);
  //     viewDetails.simulate('click');
  //     expect(history.length).toEqual(expectedCount);
  //     expect(history.location.pathname).toEqual("/pipeLineView/demoId/"+pipelineList[0]._id);
  // });
  // it("clicking edit pipeline trigger onClick event", async () => {

  //   let expectedCount = history.length + 1;
  //   const fetchpipeLineList=[{
  //     _id: "8fdc4e0aac6811e994a00242ac110019"
  //   }]

  //   const wrapper = shallow(
  //     <PipeLineList
  //       history={history}
  //       match={match}
  //       pipelineList= {fetchpipeLineList}
  //       getPipelineList={getPipelineList}
  //     />
  //   );
  //   wrapper.setProps({ pipelineList: pipelineList});
  //   let createButton = wrapper.find("button.text-primary").first();
  //   createButton.simulate("click");
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/addOrEditPipeLine/demoId/"+pipelineList[0]._id);
  // });

  //   it("clicking delete pipeline trigger onClick event", async () => {
  //     const match = {
  //       params: {
  //         id: "demoId"
  //       }
  //     };
  //     let expectedCount = history.length + 1;
  //     const wrapper = shallow(
  //       <PipeLineList
  //         history={history}
  //         match={match}
  //         pipelineList=""
  //         getPipelineList={getPipelineList}
  //       />
  //     );
  //     wrapper.setProps({ pipelineList: pipelineList});
  //     let viewDetails = wrapper.find("button.btn-danger");
  //     viewDetails.simulate("click");
  //     expect(viewDetails.length).toBe(1);
  //     expect(history.length).toEqual(expectedCount);
  //     expect(history.location.pathname).toEqual("/pipelineDeleteHandler/demoId/");
  //   });

  it("test componentWillMount", async () => {
    const match = {
      params: {
        id: "demoId"
      }
    };
    const getPipelineList = jest.fn();
    spyOn(PipeLineList.prototype, "componentWillMount").and.callThrough();
    shallow(
      <PipeLineList
        history={history}
        match={match}
        getPipelineList={getPipelineList}
        managePlantList={managePlantList}
      />
    );
    expect(PipeLineList.prototype.componentWillMount).toHaveBeenCalledTimes(1);
  });

  it("test component will recieve props", async () => {
    const getPipelineList = jest.fn();
    spyOn(
      PipeLineList.prototype,
      "componentWillReceiveProps"
    ).and.callThrough();
    const wrapper = shallow(
      <PipeLineList
        history={history}
        match={match}
        getPipelineList={getPipelineList}
        managePlantList={managePlantList}
        manageGetAlarmList={manageGetAlarmList}
        pipelineDeleteHandler={pipelineDeleteHandler}
        pipelineList=""
        pipelineListFailure=""
        pipeLineDelete=""
        pipelineDeletedSuccess=""
        pipeLineDeleteFailure=""
        plantList=""
        plantListFailure=""
        getAlarmList=""
        getAlarmListFailure=""

      />
    );
    wrapper.setProps({
      pipelineList,
      pipelineListFailure,
      pipeLineDelete,
      pipelineDeletedSuccess,
      pipeLineDeleteFailure,
      plantList,
      plantListFailure,
      getAlarmList,
      getAlarmListFailure
    });
    expect(
      PipeLineList.prototype.componentWillReceiveProps
    ).toHaveBeenCalledTimes(1);
  });

  it("Function testing on mapDispatchToProps", () => {
    const payload = "demo";
    const mockParams = {
      params: {
        plantId: "plantId",
        pipeId: "pipelineId",
        limit: "demo",
        offset: "demo"
      }
    };
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).getPipelineList(mockParams.params.plantId);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: GET_PIPELINE_LIST,
      id: "plantId"
    });

    mapDispatchToProps(dispatch).pipelineDeleteHandler(mockParams.params.plantId, mockParams.params.pipeId);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: DELETE_PIPELINE,
      plantId: "plantId",
      pipeId: "pipelineId"
    });

    mapDispatchToProps(dispatch).managePlantList();
    expect(dispatch.mock.calls[2][0]).toEqual({
      type: GET_PLANT_LIST,
    });

    mapDispatchToProps(dispatch).manageGetAlarmList(mockParams.params.plantId, mockParams.params.limit, mockParams.params.offset);
    expect(dispatch.mock.calls[3][0]).toEqual({
      type: GET_ALARM_LIST,
      plantId: "plantId",
      limit: "demo",
      offset: "demo"
    });

  });
});
