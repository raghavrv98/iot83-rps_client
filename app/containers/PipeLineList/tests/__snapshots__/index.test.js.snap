// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< PipeLineList /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      PipeLineList
    </title>
    <meta
      content="Description of PipeLineList"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb "
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Dashboard"
          id="app.common.messages.dashboard"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Pipelines"
          id="app.containers.PipeLineList.allAvailablePipelines"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30"
    >
      <div
        className="form-group mr-b-0 flex align-items-baseline"
      >
        <label
          className="form-label f-13 fx-b40 pd-r-10 text-right"
        >
          <FormattedMessage
            defaultMessage="Select Plant"
            id="app.common.messages.selectPlant"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
           :
        </label>
        <StateManager
          className="form-control-multi-select fx-b60"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="selectedPlant"
          isDisabled={true}
          onChange={[Function]}
          options={Array []}
          value={
            Object {
              "label": "ALL",
              "value": "all",
            }
          }
        />
      </div>
    </div>
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonFullView"
  />
</div>
`;
